base/serialize.jl

```diff
@@ -102,14 +102,20 @@ function serialize_cycle(s::SerializationState, x)
     return false
 end
 
-serialize(s::SerializationState, x::Bool) = x ? writetag(s.io, TRUE_TAG) :
+function reset(s::AbstractSerializer)
```

Comment:
The current one works on streams - ` Reset a stream s to a previously marked position, and remove the mark. Returns the previously marked position. Throws an error if the stream is not marked.`

Here it helps in reusing `Serializer` objects and is similar in intent.

Changing `reset` to `reset!` should be discussed in a separate issue since it will be quite widely used in packages (in the context of streams).